lib/liquid/nimble_parser.ex

```diff
@@ -144,14 +141,19 @@ defmodule Liquid.NimbleParser do
   )
 
   @doc """
-  Validates and parse liquid markup
+  Validates and parse liquid markup.
   """
   @spec parse(String.t()) :: {:ok | :error, any()}
+  def parse(""), do: {:ok, []}
+
   def parse(markup) do
     case __parse__(markup) do
       {:ok, template, "", _, _, _} ->
         {:ok, template}
 
+      {:error, mesagge, _, _, _, _} ->
+        {:error, "#{mesagge}"}
+
       {:ok, _, rest, _, _, _} ->
         {:error, "Error parsing: #{rest}"}
```

Comment:
What is `rest` in this case?

Reply:
Rest contains all markup from error raises. It contains where error was found. (Could not be parsed)

Reply:
Renamed: `rest` -> `incorrect_markup`